XamarinCommunityToolkit/Views/GravatarImageSource/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,136 @@
+
+		public static readonly BindableProperty DefaultProperty =
+			BindableProperty.Create(nameof(Default), typeof(DefaultGravatar), typeof(GravatarImageSource), DefaultGravatar.MysteryPerson, propertyChanged: OnGravatarPropertyChanged);
+
+		public static readonly BindableProperty CachingEnabledProperty =
+			BindableProperty.Create(nameof(CachingEnabled), typeof(bool), typeof(GravatarImageSource), true, propertyChanged: OnGravatarPropertyChanged);
+
+		public static readonly BindableProperty CacheValidityProperty =
+			BindableProperty.Create(nameof(CacheValidity), typeof(TimeSpan), typeof(GravatarImageSource), TimeSpan.FromDays(7), propertyChanged: OnGravatarPropertyChanged);
+
+		static void OnGravatarPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			if (bindable is GravatarImageSource gis)
+			{
+				gis.OnSourceChanged();
+			}
+		}
+
+		public string Email
+		{
+			get => (string)GetValue(EmailProperty);
+			set => SetValue(EmailProperty, value);
+		}
+
+		public int Size
+		{
+			get => (int)GetValue(SizeProperty);
+			set => SetValue(SizeProperty, value);
+		}
+
+		public DefaultGravatar Default
+		{
+			get => (DefaultGravatar)GetValue(DefaultProperty);
+			set => SetValue(DefaultProperty, value);
+		}
+
+		public bool CachingEnabled
+		{
+			get => (bool)GetValue(CachingEnabledProperty);
+			set => SetValue(CachingEnabledProperty, value);
+		}
+
+		public TimeSpan CacheValidity
+		{
+			get => (TimeSpan)GetValue(CacheValidityProperty);
+			set => SetValue(CacheValidityProperty, value);
+		}
+
+#if !NETSTANDARD1_0
+		internal bool UseCacheFile(FileInfo file)
+		{
+			if (!file.Directory.Exists)
+				file.Directory.Create();
+
+			return CachingEnabled && file.Exists && file.CreationTime.Add(CacheValidity) > DateTime.Now;
+		}
+#endif
+
+		internal string CacheFileName(float scale) => $"{GetMd5Hash(Email)}-{Size}@{scale}x.png";
+
+		internal async Task<byte[]> GetGravatarAsync(float scale)
+		{
+#if NETSTANDARD1_0
+			throw new NotImplementedException();
+#else
+			using var client = new HttpClient();
+			var response = await client.GetAsync(GetGravatarUri(Email, scale)).ConfigureAwait(false);
+
+			if (!response.IsSuccessStatusCode)
+			{
+				return Array.Empty<byte>();
+			}
+
+			return await response.Content.ReadAsByteArrayAsync().ConfigureAwait(false);
+#endif
+		}
+
+		const string requestUriFormat = "https://www.gravatar.com/avatar/{0}?s={1}&d={2}";
+
+		string GetGravatarUri(string email, float scale) => string.Format(requestUriFormat, GetMd5Hash(email), Size * scale, DefaultGravatarName());
+
+		string DefaultGravatarName()
+		{
+			return Default switch
+			{
+				DefaultGravatar.FileNotFound => "404",
+				DefaultGravatar.MysteryPerson => "mp",
+				_ => $"{Default}".ToLower(),
+			};
+		}
+
+		string GetMd5Hash(string str)
+		{
+#if NETSTANDARD1_0
+			throw new NotImplementedException();
+#else
+			byte[] hash = null;
+			using (var md5 = MD5.Create())
```

Comment:
Can you use the C#8 style? For the HttpClient implementation you use the C#8, would be good to follow the same pattern.